ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -104,6 +106,14 @@ public AlterTableCompactOperation(DDLOperationContext context, AlterTableCompact
                 compactionRequest, ServerUtils.hostname(), txnHandler, context.getConf());
         parseCompactionResponse(compactionResponse, table, partitionMapEntry.getKey());
       }
+      // If Iceberg table had partition evolution, it will create compaction request without partition specification,
+      // and it will compact all files from old partition specs, besides compacting partitions of current spec in parallel.
+      if (DDLUtils.isIcebergTable(table) && table.getStorageHandler().isPartitioned(table) &&
+          table.getStorageHandler().isUndergonePartitionEvolution(table)) {
+        compactionRequest.setPartitionname(null);
```

Comment:
isn't that a default , null?

Reply:
Yes, but it reuses the same compaction request object when submitting compaction requests for different partitions modifying the compaction name. Existing ACID compaction functionality. Setting partition name to null because in that code block it is not null, but may have the value of the previous partition that was requested.